recipes/assimp/5.x/conanfile.py

```diff
@@ -146,66 +163,56 @@ def _depends_on_clipper(self):
 
     @property
     def _depends_on_stb(self):
-        if Version(self.version) < "5.1.0":
-            return False
         return self.options.with_m3d or self.options.with_m3d_exporter or \
             self.options.with_pbrt_exporter
 
-    @property
-    def _depends_on_zlib(self):
-        return self.options.with_assbin or self.options.with_assbin_exporter or \
-            self.options.with_assxml_exporter or self.options.with_blend or self.options.with_fbx or \
-            self.options.with_q3bsp or self.options.with_x or self.options.with_xgl
-
     @property
     def _depends_on_openddlparser(self):
-        if Version(self.version) < "5.1.0":
-            return False
         return self.options.with_opengex
 
     def requirements(self):
         # TODO: unvendor others libs:
         # - Open3DGC
         self.requires("minizip/1.2.13")
-        self.requires("utfcpp/3.2.3")
-        if Version(self.version) < "5.1.0":
-            self.requires("irrxml/1.2")
-        else:
-            self.requires("pugixml/1.13")
+        self.requires("pugixml/1.14")
+        self.requires("utfcpp/4.0.1")
+        self.requires("zlib/[>=1.2.11 <2]")
```

Comment:
Why don't you keep the same condition for zlib as before? I don't understand the rational explained in your first message.

Reply:
While the existing logic was valid, I found it to mostly add unnecessary maintenance overhead. There is not much point in making `zlib` an optional dependency when it's the most common dependency around and also already added to the build via the non-optional `minizip` dependency. I don't think reviewing the project to keep track of which components exactly need or don't need `zlib` is worth the effort when bumping the package version.

Reply:
zlib may not be a direct dependency, maintenance overhead is orthogonal.

Distinction between transitive dependencies and direct dependencies is important, specially in conan v2.

Reply:
Yes, but that's just one of the reasons. I'm still not convinced that keeping `zlib` optional is a worthwhile endeavor.